src/ripple/protocol/STArray.h

```diff
@@ -71,11 +71,35 @@ class STArray final
         return emplace(n, buf, std::move(*this));
     }
 
-    void push_back (const STObject & object)
+    STObject& operator[] (std::size_t j)
+    {
+        return v_[j];
+    }
+
+    STObject const& operator[] (std::size_t j) const
+    {
+        return v_[j];
+    }
+
+    STObject& back() { return v_.back(); }
```

Comment:
Is similarity to `vector` worth having? There is currently no use-case for `front`. Existing code calls `push_back` followed by a call to `back` to get a reference to perform subsequent modifications. Were we to do this using `push_front` followed by `front`, it would be less efficient (since the entire set of existing elements would always need to be moved). I feel that we should add `front` only when there is a compelling reason.